packages/overlay/src/integrations/sentry/data/sentryStore.ts

```diff
@@ -247,32 +265,34 @@ class SentryDataCache {
       else trace.rootTransactionName = '(missing root transaction)';
 
       if (!existingTrace) {
-        this.traces.unshift(trace);
-        this.tracesById.set(trace.trace_id, trace);
-      }
-
-      for (const [type, cb] of this.subscribers.values()) {
-        if (type === 'trace') {
-          cb(trace);
-        }
+        const newTracesById = new Map(tracesById);
+        newTracesById.set(trace.trace_id, trace);
+        set({
+          traces: [trace, ...traces],
```

Comment:
This way of creating a new array is quite wasteful so let's go back to `unshift`?